java/ingest/ingest-runner/src/test/java/sleeper/ingest/impl/IngestCoordinatorCommonIT.java

```diff
@@ -595,59 +787,95 @@ public void shouldWriteDuplicateRecords(
                 recordListAndSchema.sleeperSchema);
-        Function<Key, Integer> keyToPartitionNoMappingFn = key -> 0;
-        Map<Integer, Integer> partitionNoToExpectedNoOfFilesMap = Stream.of(
-                        new AbstractMap.SimpleEntry<>(0, 1))
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
+        DynamoDBStateStore stateStore = new DynamoDBStateStoreCreator(UUID.randomUUID().toString(), duplicatedRecordListAndSchema.sleeperSchema, AWS_EXTERNAL_RESOURCE.getDynamoDBClient()).create();
 
-        PartitionTree tree = new PartitionsBuilder(recordListAndSchema.sleeperSchema)
+        PartitionTree tree = new PartitionsBuilder(duplicatedRecordListAndSchema.sleeperSchema)
                 .rootFirst("root")
                 .buildTree();
+        stateStore.initialise(tree.getAllPartitions());
+        Instant stateStoreUpdateTime = Instant.parse("2023-08-08T11:20:00Z");
+        String ingestLocalWorkingDirectory = createTempDirectory(temporaryFolder, null).toString() + "/path/to/new/sub/directory";
+        IngestCoordinatorTestParameters parameters = createTestParameterBuilder()
+                .fileNames(List.of("leftFile", "rightFile"))
+                .fileUpdatedTimes(List.of(stateStoreUpdateTime, stateStoreUpdateTime, stateStoreUpdateTime))
+                .stateStore(stateStore)
+                .schema(duplicatedRecordListAndSchema.sleeperSchema)
+                .workingDir(ingestLocalWorkingDirectory)
+                .build();
+
+        // When`
+        ingestRecords(duplicatedRecordListAndSchema, parameters, ingestType);
+
+        // Then
+        List<FileInfo> actualFiles = stateStore.getActiveFiles();
+        List<Record> actualRecords = readMergedRecordsFromPartitionDataFiles(duplicatedRecordListAndSchema.sleeperSchema, actualFiles, hadoopConfiguration);
+        FileInfoFactory fileInfoFactory = FileInfoFactory.builder()
+                .partitionTree(tree)
+                .lastStateStoreUpdate(stateStoreUpdateTime)
+                .schema(duplicatedRecordListAndSchema.sleeperSchema)
+                .build();
+        List<FileInfo> fileInfoList = List.of(
+                fileInfoFactory.leafFile(ingestType.getFilePrefix(parameters) + "/partition_root/leftFile.parquet", 400, -100L, 99L)
+        );
+        assertThat(Paths.get(ingestLocalWorkingDirectory)).isEmptyDirectory();
+        assertThat(actualFiles).containsExactlyInAnyOrderElementsOf(fileInfoList);
+        assertThat(actualRecords).containsExactlyInAnyOrderElementsOf(duplicatedRecordListAndSchema.recordList);
+        List<List<Object>> recordList = LongStream.range(-100, 100).boxed().map(longValue -> List.of((Object) longValue)).collect(Collectors.toList());
+        recordList.addAll(recordList);
+        assertThat(actualRecords).extracting(record -> record.getValues(List.of("key0")))
+                .containsExactlyInAnyOrderElementsOf(recordList);
```

Comment:
It looks like this should be `containsExactlyElementsOf` instead of InAnyOrder. You could generate a sorted list by using `LongStream.range(-100, 100).boxed().flatMap(longValue -> Stream.of(longValue, longValue))` rather than using addAll.